src/components/Button.js

```diff
@@ -1,23 +1,91 @@
-import React from "react";
-import { TouchableOpacity, StyleSheet, Text } from "react-native";
+import React from 'react';
+import { TouchableOpacity, StyleSheet, Text, Dimensions } from 'react-native';
+
+const screen = Dimensions.get("window");
+const buttonWidth = screen.width / 4;
 
 const styles = StyleSheet.create({
   text: {
-    color: '#000',
-    fontSize: 24,
-    textAlign: 'center',
+    color: '#CBE1FF',
+    fontSize: 34,
   },
   button: {
-    minWidth: '20%',
-    minHeight: '40%',
-    padding: '5%',
-    backgroundColor: 'white',
+    flex: 1,
+    height: Math.floor(buttonWidth - 10),
+    backgroundColor: '#242933',
+    borderRadius: 2,
+    alignItems: 'center',
+    justifyContent: 'center',
+    margin: 5,
+  },
+  buttonDouble: {
+    // width: Math.floor(screen.width / 2 - 20),
+    width: buttonWidth * 2 - 18,
+    flex: 0,
+    alignItems: 'flex-start',
+    paddingLeft: 40,
+  },
+  secondary_button: {
+    backgroundColor: '#363E4D',
+  },
+  primary_button: {
+    backgroundColor: '#0060E5',
+  },
+  topLeftCorner: {
+    borderTopLeftRadius: 8,
+    borderTopRightRadius: 2,
+    borderBottomRightRadius: 2,
+    borderBottomLeftRadius: 2,
+  },
+  topRightCorner: {
+    borderTopLeftRadius: 2,
+    borderTopRightRadius: 8,
+    borderBottomRightRadius: 2,
+    borderBottomLeftRadius: 2,
+  },
+  bottomLeftCorner: {
+    borderTopLeftRadius: 2,
+    borderTopRightRadius: 2,
+    borderBottomRightRadius: 2,
+    borderBottomLeftRadius: 8,
+  },
+  bottomRightCorner: {
+    borderTopLeftRadius: 2,
+    borderTopRightRadius: 2,
+    borderBottomRightRadius: 8,
+    borderBottomLeftRadius: 2,
   },
 });
 
-export default ({ text, onPress }) => {
+export default ({ text, onPress, size, theme }) => {
+  const buttonStyles = [styles.button];
+  if (size === 'double') {
+    buttonStyles.push(styles.buttonDouble);
+  }
+  if (theme) {
```

Comment:
I think this could be a switch statement.

Reply:
Did you mean inside the foreach loop? I think if-else statements are fine. If you think using switch case is necessary then we can adjust it later on.